lib/puppet/defaults.rb

```diff
@@ -786,6 +786,20 @@ def self.default_diffargs
         :values => ["md5", "sha256"],
         :desc     => 'Which digest algorithm to use for file resources and the filebucket.
                       Valid values are md5, sha256. Default is md5.',
+    },
+    :supported_checksum_types => {
+      :default => ['md5', 'sha256'],
+      :type    => :array,
+      :desc    => 'Checksum types supported by this agent for use in file resources of a static
+        catalog. Valid types are md5, md5lite, sha256, sha256lite, sha1, sha1lite, mtime, ctime.',
+      :hook    => proc do |value|
+        values = munge(value)
+        valid   = ['md5', 'md5lite', 'sha256', 'sha256lite', 'sha1', 'sha1lite', 'mtime', 'ctime']
```

Comment:
We have the same issue with [log levels](https://github.com/MikaelSmith/puppet/blob/PUP-5120/lib/puppet/defaults.rb#L87) so I'm ok not trying to solve that here.